src/base_output_updater.cpp

```diff
@@ -36,6 +36,57 @@ void compute_delta_z_output(std::vector<float> &mu_a, std::vector<float> &var_a,
     }
 }
 
+void compute_delta_z_noise(std::vector<float> &mu_a, std::vector<float> &var_a,
+                           std::vector<float> &jcb, std::vector<float> &obs,
+                           int start_chunk, int end_chunk,
+                           std::vector<float> &delta_mu,
+                           std::vector<float> &delta_var) {
+    const float zero_pad = 0.0f;
+
```

Comment:
I would add comment here to explain the ordering of the output vector `mu_a` and `var_a`